extensions/integration-tests/src/notebook.test.ts

```diff
@@ -14,136 +14,183 @@ import { sqlNotebookContent, writeNotebookToFile, sqlKernelMetadata, getFileName
+	async sqlNbMultipleCellsTest(title: string): Promise<void> {
+		let notebook = await this.openNotebook(sqlNotebookMultipleCellsContent, sqlKernelMetadata, title + this.invocationCount++, true);
+		const expectedOutput0 = '(1 row affected)';
+		for (let i = 0; i < 3; i++) {
+			let cellOutputs = notebook.document.cells[i].contents.outputs;
+			console.log(`Got cell outputs --- ${i}`);
+			if (cellOutputs) {
+				cellOutputs.forEach(o => console.log(o));
+			}
+			assert(cellOutputs.length === 3, `Expected length: 3, Actual: '${cellOutputs.length}'`);
+			let actualOutput0 = (<azdata.nb.IDisplayData>cellOutputs[0]).data['text/html'];
+			console.log('Got first output');
+			assert(actualOutput0 === expectedOutput0, `Expected row count: '${expectedOutput0}', Actual: '${actualOutput0}'`);
+			let actualOutput2 = (<azdata.nb.IExecuteResult>cellOutputs[2]).data['application/vnd.dataresource+json'].data[0];
+			assert(actualOutput2[0] === i.toString(), `Expected result: ${i.toString()}, Actual: '${actualOutput2[0]}'`);
+			console.log('Sql multiple cells NB done');
+		}
+	}
+
+	@stressify({ dop: NotebookTester.ParallelCount, iterations: NotebookTester.IterationCount })
+	async sqlNbTest(title: string): Promise<void> {
+		let notebook = await this.openNotebook(sqlNotebookContent, sqlKernelMetadata, title + this.invocationCount++, false, true);
+		const expectedOutput0 = '(1 row affected)';
+		let cellOutputs = notebook.document.cells[0].contents.outputs;
+		console.log('Got cell outputs ---');
+		if (cellOutputs) {
+			cellOutputs.forEach(o => console.log(o));
+		}
+		assert(cellOutputs.length === 3, `Expected length: 3, Actual: ${cellOutputs.length}`);
+		let actualOutput0 = (<azdata.nb.IDisplayData>cellOutputs[0]).data['text/html'];
+		console.log('Got first output');
+		assert(actualOutput0 === expectedOutput0, `Expected row count: ${expectedOutput0}, Actual: ${actualOutput0}`);
+		let actualOutput2 = (<azdata.nb.IExecuteResult>cellOutputs[2]).data['application/vnd.dataresource+json'].data[0];
+		assert(actualOutput2[0] === '1', `Expected result: 1, Actual: '${actualOutput2[0]}'`);
+	}
+
+	async cleanup(testName: string): Promise<void> {
+		try {
+			let fileName = getFileName(testName + this.invocationCount++);
+			if (fs.existsSync(fileName)) {
+				fs.unlinkSync(fileName);
+				console.log(`"${fileName}" is deleted.`);
+			}
+			await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+		}
+		catch (err) {
+			console.log(err);
+		}
+		finally {
+			console.log(`"${testName}" is done`);
+		}
+	}
+
+	async openNotebook(content: azdata.nb.INotebookContents, kernelMetadata: any, testName: string, runAllCells?: boolean, connectToDifferentServer?: boolean): Promise<azdata.nb.NotebookEditor> {
+		let notebookConfig = vscode.workspace.getConfiguration('notebook');
+		notebookConfig.update('pythonPath', getConfigValue(EnvironmentVariable_PYTHON_PATH), 1);
+		if (!connectToDifferentServer) {
+			let server = await getBdcServer();
```

Comment:
I don't get what this is doing - you don't appear to actually be connecting to a different server when this is true? And do you need to connect each time to this server? You added a call to connect in the setup function - isn't that enough for these tests?

Reply:
this is not new code, this method pre-existed, I did not look into it and treated it as a black box. The call to the connedct in the setup function should indeed be enough. It was added by Chris and i reckon he would come back and clean this up. I will check with him before merge to figure how he would like to proceed.

Reply:
I will work with Chris to figure how he wants to handle this.

---
In reply to: [287894810](https://github.com/microsoft/azuredatastudio/pull/5499#discussion_r287894810) [](ancestors = 287894810)

Reply:
This was a fix that was done for the stress tests to ensure that we don't continue connecting to a new server. probably should be named something like 'isConnectionNecessary' or something like that. I can fix in an upcoming PR.